core/core.go

```diff
@@ -256,26 +270,49 @@ func (c *Core) RequestDomToAppendOrFetch(hash common.Hash, order int) {
 		_, exists := c.appendQueue.Get(hash)
 		if !exists {
 			log.Warn("Block sub asked doesnt exist in append queue, so request the peers for it", "Hash", hash, "Order", order)
-			c.sl.missingParentFeed.Send(hash) // Using the missing parent feed to ask for the block
+			block := c.GetBlockOrCandidateByHash(hash)
+			if block == nil {
+				c.sl.missingBlockFeed.Send(types.BlockRequest{Hash: hash, Entropy: entropy}) // Using the missing parent feed to ask for the block
+			} else {
+				// Check if the hash is in the blockchain, otherwise add it to the append queue
+				if c.GetHeaderByHash(block.Hash()) == nil {
+					c.addToAppendQueue(block)
+				}
+			}
 		}
 	} else if nodeCtx == common.REGION_CTX {
 		if order < nodeCtx { // Prime block
 			if c.sl.domClient != nil {
-				go c.sl.domClient.RequestDomToAppendOrFetch(context.Background(), hash, order)
+				go c.sl.domClient.RequestDomToAppendOrFetch(context.Background(), hash, entropy, order)
 			}
 		}
 		_, exists := c.appendQueue.Get(hash)
 		if !exists {
 			log.Warn("Block sub asked doesnt exist in append queue, so request the peers for it", "Hash", hash, "Order", order)
-			c.sl.missingParentFeed.Send(hash) // Using the missing parent feed to ask for the block
+			block := c.GetBlockByHash(hash)
+			if block == nil {
+				c.sl.missingBlockFeed.Send(types.BlockRequest{Hash: hash, Entropy: entropy}) // Using the missing parent feed to ask for the block
+			} else {
+				// Check if the hash is in the blockchain, otherwise add it to the append queue
+				if c.GetHeaderByHash(block.Hash()) == nil {
+					c.addToAppendQueue(block)
+				}
+			}
 		}
 	}
 
 }
 
 // addToAppendQueue adds a block to the append queue
 func (c *Core) addToAppendQueue(block *types.Block) error {
-	c.appendQueue.ContainsOrAdd(block.Hash(), blockNumberAndRetryCounter{block.NumberU64(), 0})
+	nodeCtx := common.NodeLocation.Context()
+	_, order, err := c.engine.CalcOrder(block.Header())
+	if err != nil {
+		return err
+	}
+	if order == nodeCtx {
+		c.appendQueue.ContainsOrAdd(block.Hash(), blockNumberAndRetryCounter{block.NumberU64(), 0})
```

Comment:
Idk what that gives us? we are violating the lru cache ? it takes in any interface. Since this header already been verified, i dont see it being useful